### DISCUSSION TITLE
Appending multiple particle set outputs to the same .Xarr file

### DISCUSSION THREAD

**Discussion by TomMBIO** [Parcels-code/Parcels#1731] [General]
Hi everyone,

I have made some progress with trying to run multiple particle sets, each beginning on a subsequent week:

`
iterations = range(26)  # Simulate for 26 weeks
psets = [] 

cohort = 0  
release_time = datetime.datetime(2022, 3, 1, 00, 00, 0)

for i in iterations:
    cohort_array = np.full(n_sites, cohort) #create an array of cohort values for each site
    # Create a new ParticleSet for each iteration with updated release time and cohort ID
    pset = ParticleSet(
        fieldset=fieldset,
        pclass=Particle,
        lat=release_lat,
        lon=release_lon,
        depth = None,
        time=release_time, 
        cohort=cohort_array
    )
    
    psets.append(pset)

    kernels = pset.Kernel(Sample_land) + pset.Kernel(Unbeaching) + pset.Kernel(AdvectionRK4) + pset.Kernel(DiffusionUniformKh) + pset.Kernel(SampleT)
    output_file = pset.ParticleFile(name=r"<filepath>\mytilus1_6.zarr", outputdt=datetime.timedelta(minutes=60), write_mode='append')
    
    # Execute the simulation for this ParticleSet
    pset.execute(
        kernels,
        runtime=datetime.timedelta(days=80), 
        dt=datetime.timedelta(minutes=60),
        output_file=output_file  
    )
    
    # Increment the cohort ID and release time for the next batch of particles
    cohort += 1
    release_time += datetime.timedelta(days=7)
`

Just to clarify, I am doing this instead of using repeatdt because I wish to assign each release event a cohort number.

It appears **write_mode='append' does not have functionality with parcels**, and I am a bit stuck. One solution could be to assign each output its own file, and combine them in post, but this feels a bit unsophisticated. Do you have any ways to append trajectories to the same xarr file automatically? I assume using multiple particlesets for an experiment is common practice but not sure.

Thank you,
Tom

**Comment by erikvansebille** [comment]
Hi @TomMBIO, I don't think you need `write_mode='append'`. Appending is the default (and only) way of writing to a zarr file in Parcels. What happens if you don't give that argument?

**Comment by TomMBIO** [comment]
Hi Erik, 

When I didn't do this, the particle files appeared to be overwritten with each iteration. But I will double check this

**Reply by erikvansebille** [reply]
Ah, you'll have to move the creation of the ParticleFile object outside of the `for`-loop.

**Reply by TomMBIO** [reply]
This appears to have worked. Thank you!
